### PR TITLE
New AO5GC 1.0 service - 2023-05-15-preview

### PR DESCRIPTION
**Pull request by satravi**
# ARM (Control Plane) API Specification Update Pull Request 

## PR review workflow diagram

Please understand this diagram before proceeding. It explains how to get your PR approved & merged.

diagram

## Purpose of this PR

What's the purpose of this PR? Check all that apply. This is **mandatory**!

  - [X] New service onboarding
  - [ ] New API version. (Such PR should have been generated with [OpenAPI Hub](https://aka.ms/openapihub), per [this wiki doc](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).)
  - [ ] Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
  - [ ] Update existing version to fix swagger quality issues in S360.
  - [ ] Other, please clarify:
    - _edit this with your clarification_

## Due diligence checklist

To merge this PR, you **must** go through the following checklist and confirm you understood 
and followed the instructions by checking all the boxes:

- [X] I have reviewed the general guidance on the spec PR review process: https://aka.ms/specprreview.
- [X] I confirm this PR is modifying Azure Resource Manager (ARM) related specifications, and not data-plane related specifications.
- [X] I commit to follow the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy).
- [X] I have reviewed following [Resource Provider guidelines](https://aka.ms/rpguidelines), including
  [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and
  [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md) (estimated time: 4 hours).  
  I understand this is required before I can request review from an ARM API Review board.

### ARM API changes review

- If you want for the ARM team to review this PR, you must add the `ARMReview` label. 
- The automation may automatically add the `ARMReview` label, if appropriate.  
  If this happens, proceed according to guidance given in GitHub comments also added by the automation.

### Breaking change review

If you have any breaking changes as defined in the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy/), 
follow the process outlined in the [High-level Breaking Change Process doc](https://aka.ms/breakingchangesprocess#high-level-breaking-change-process).
      
## Getting help

- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- For additional help, see https://aka.ms/azsdk/support/spectools.

### CODE REVIEW

**Comment by satravi** [comment]
This is Pull Request for a new service - MobilePacketCore mpcnetworkfunction.
Below are the **ARMSignedOff** PRs from the Private Repo - RPSaaSMaster for the PR changes

https://github.com/Azure/azure-rest-api-specs-pr/pull/12558
https://github.com/Azure/azure-rest-api-specs-pr/pull/12921
https://github.com/Azure/azure-rest-api-specs-pr/pull/13648

**Review by mikeharder — DISMISSED** [review]
Check "specs - typespec - pr" is failing because your `*.tsp` files are required to be formatted using the `tsp format` command.

**Comment by satravi** [comment]
@mikeharder We have resolved the *.tsp file format errors.
Currently 'Service API Readiness Test' is failing. Found the below note in the doc
"Note: Currently only applicable to management plane APIs, and target ARM region is US West Central - the SDP pilot region."

So this is not applicable for our service.

**Comment by mgl1969** [comment]
@mikeharder - we're getting the following error in the APIView check.  It indicates to ask for help from the PR assignee.

2023-07-27T13:16:09.0737240Z {"level":"error","message":"Encounter unexpected Error when creating ApiView for mpcnetworkfunction.Management"}
2023-07-27T13:16:09.0740615Z Publishing event to eventHub: {
2023-07-27T13:16:09.0741083Z   status: 'apiview',
2023-07-27T13:16:09.0741394Z   env: 'prod',
2023-07-27T13:16:09.0741689Z   source: 'github',
2023-07-27T13:16:09.0742002Z   unifiedPipelineTaskKey: 'ApiView',
2023-07-27T13:16:09.0742385Z   unifiedPipelineBuildId: '1ddf6b45-b09d-4bc1-a72a-26b328c8d94d',
2023-07-27T13:16:09.0742674Z   apiViewLink: undefined,
2023-07-27T13:16:09.0743032Z   errorMessage: 'Create ApiView failed. Please ask PR assignee for help',
2023-07-27T13:16:09.0743360Z   language: 'TypeSpec',
2023-07-27T13:16:09.0743679Z   packageName: 'mpcnetworkfunction.Management'
2023-07-27T13:16:09.0743942Z }
2023-07-27T13:16:09.0744472Z {"level":"info","message":"EventHubProducer.constructor: connectionString: Endpoint=sb://openapi-platform-events.servicebus.windows.net/;SharedAccessKeyName=pipeline-app-send;SharedAccessKey=*****;EntityPath=pipeline-events."}
2023-07-27T13:16:09.0745563Z {"level":"info","message":"EventHubProducer.send: FQN/eventHubName: openapi-platform-events.servicebus.windows.net/pipeline-events, events: {\"status\":\"apiview\",\"env\":\"prod\",\"source\":\"github\",\"unifiedPipelineTaskKey\":\"ApiView\",\"unifiedPipelineBuildId\":\"1ddf6b45-b09d-4bc1-a72a-26b328c8d94d\",\"errorMessage\":\"Create ApiView failed. Please ask PR assignee for help\",\"language\":\"TypeSpec\",\"packageName\":\"mpcnetworkfunction.Management\"}"}
2023-07-27T13:16:10.2525629Z {"level":"info","message":"EventHubProducer.send: FQN/eventHubName: openapi-platform-events.servicebus.windows.net/pipeline-events, Done"}
2023-07-27T13:16:10.6484504Z ##[error]Bash exited with code '1'.
2023-07-27T13:16:10.6492517Z ##[section]Finishing: Generate ApiView
2023-07-27T13:16:10.6520470Z ##[section]Starting: After TypeSpecAPIView

**Comment by mgl1969** [comment]
@mikeharder - this is for an RPSaaS based RP.  The service API readiness test is failing due to this.

API Readiness Check

This CI check is to make sure service is ready before PR merge. Technically, the CI check send operationsList HTTP request to Azure Resource Provider.

To fix this CI check failure, if you haven't got ARM signed off, pls get ARM signed off first then deploy ARM manifest. After deploying ARM manifest, this operationsList HTTP request will succeed and CI pass.

NOTE: If your RP is RPaaS RP, since RPaaS requires swagger merge first. In this case, you could ignore this CI check.

File: specification/mpcnetworkfunction/resource-manager/Microsoft.MobilePacketCore/preview/2023-05-15-preview/networkfunction.json
Comment (visingla-ms) [inline specification/mpcnetworkfunction/resource-manager/Microsoft.MobilePacketCore/preview/2023-05-15-preview/networkfunction.json:50]: >Microsoft.MobilePacketCore [](http://example.com/codeflow?start=16&length=26)

Did you get the review and approval done for your new RP Namespace name? See: [https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/resource-api-reference.md#resource-provider-and-type-names-best-practices](https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/resource-api-reference.md#resource-provider-and-type-names-best-practices).

Comment (satravi) [reply specification/mpcnetworkfunction/resource-manager/Microsoft.MobilePacketCore/preview/2023-05-15-preview/networkfunction.json:50]: @visingla-ms 
Yes, we have got the **ARMSignedOff** for the Namespace and the Swagger APIs. Please refer the below PR
https://github.com/Azure/azure-rest-api-specs-pr/pull/12558

File: specification/mpcnetworkfunction/resource-manager/readme.typescript.md
Comment (qiaozha) [inline specification/mpcnetworkfunction/resource-manager/readme.typescript.md:10]:
```suggestion
  package-name: "@azure/arm-mobilecore"
  output-folder: "$(typescript-sdks-folder)/sdk/mobilecore/arm-mobilecore"
```

**Comment by satravi** [comment]
@qiaozha Thank you. Have pushed the suggested changes to the PR. Could you please review and approve?

**Comment by qiaozha** [comment]
> @qiaozha Thank you. Have pushed the suggested changes to the PR. Could you please review and approve?

we will need ARM sign-off, I am only responsible for the JS SDK part review.

File: specification/mpcnetworkfunction/mpcnetworkfunction.Management/tspconfig.yaml
Comment (mikeharder) [inline specification/mpcnetworkfunction/mpcnetworkfunction.Management/tspconfig.yaml:7]:
```suggestion
emit:
  - "@azure-tools/typespec-autorest"
options:
  "@azure-tools/typespec-autorest":
    azure-resource-provider-folder: "resource-manager"
    emitter-output-dir: "{project-root}/.."
    examples-directory: "examples"
    output-file: "{azure-resource-provider-folder}/{service-name}/{version-status}/{version}/networkfunction.json"
```

See this spec as an example:

https://github.com/Azure/azure-rest-api-specs/blob/9025f6894e0df9ebbfcd19a04c967c1c6737bfe6/specification/servicenetworking/ServiceNetworking.Management/tspconfig.yaml#L1-L8

**Review by mikeharder — DISMISSED** [review]

**Comment by mikeharder** [comment]
> @mikeharder - we're getting the following error in the APIView check. It indicates to ask for help from the PR assignee.
> 
> 2023-07-27T13:16:09.0737240Z {"level":"error","message":"Encounter unexpected Error when creating ApiView for mpcnetworkfunction.Management"} 2023-07-27T13:16:09.0740615Z Publishing event to eventHub: { 2023-07-

I have reported this to the owners of APIView (@praveenkuttappan, @chidozieononiwu, @maririos).

**Comment by apalanichamy1981** [comment]
Hi @mikeharder  -  Can you please shed light on this? This approval takes longer time, our private preview should go out today.  It would be great if you could share what are pending, what is from our side?

Also, the Service API readiness test is not required as we use RPaaS - RP design in our service. Our RP is sitting behind the RPaaS. Please help us on this. If required, we can have a quick call to go over and close this.

**Comment by satravi** [comment]
Thank you @mikeharder 
We have addressed the TypeSpec validation errors.
Now all the required validation checks completed successfully. Could you please review and approve the changes?

**Review by rkmanda — APPROVED** [review]

**Pull request merged**
